### PR TITLE
Enforce timeouts in Faraday's test adapter

### PR DESCRIPTION
**Pull request by dpep**
## Description
Upgrade faraday's test adapter to enforce connection timeouts.  Inspired by the [faraday-rack](https://github.com/lostisland/faraday-rack/blob/main/lib/faraday/adapter/rack.rb) implementation.

### CODE REVIEW

File: lib/faraday/adapter.rb [hunk 1]
```diff
@@ -78,8 +78,7 @@ def save_response(env, status, body, headers = nil, reason_phrase = nil, finishe
     # @param type [Symbol] Describes which timeout setting to get: :read,
     #                      :write, or :open.
     # @param options [Hash] Hash containing Symbol keys like :timeout,
-    #                       :read_timeout, :write_timeout, :open_timeout, or
-    #                       :timeout
+    #                       :read_timeout, :write_timeout, or :open_timeout
```
Comment (dpep) [inline lib/faraday/adapter.rb:81]: dedup `:timeout` which is also on line 80

File: lib/faraday/adapter.rb [hunk 2]
```diff
@@ -78,8 +78,7 @@ def save_response(env, status, body, headers = nil, reason_phrase = nil, finishe
     # @param type [Symbol] Describes which timeout setting to get: :read,
     #                      :write, or :open.
     # @param options [Hash] Hash containing Symbol keys like :timeout,
-    #                       :read_timeout, :write_timeout, :open_timeout, or
-    #                       :timeout
```
Comment (dpep) [inline lib/faraday/adapter.rb:82]: dedup `:timeout` which is also on line 80

**Comment by olleolleolle** [comment]
(This is a super-cool PR, @dpep, the tests run after linting passes.)

<details>

```
Offenses:

lib/faraday/adapter/test.rb:283:11: C: [Correctable] Layout/IndentationWidth: Use 2 (not -7) spaces for indentation. ([https://rubystyle.guide#spaces-indentation](https://rubystyle.guide/#spaces-indentation))
          [env, meta].take(block_arity)
          ^^^^^^^
lib/faraday/adapter/test.rb:284:9: C: [Correctable] Layout/ElseAlignment: Align else with if.
        else
        ^^^^
lib/faraday/adapter/test.rb:286:9: W: [Correctable] Layout/EndAlignment: end at 286, 8 is not aligned with if at 282, 17.
        end
        ^^^
lib/faraday/adapter/test.rb:300:1: C: [Correctable] Layout/EmptyLines: Extra blank line detected. ([https://rubystyle.guide#two-or-more-empty-lines](https://rubystyle.guide/#two-or-more-empty-lines))

75 files inspected, 4 offenses detected, 4 offenses autocorrectable
Error: Process completed with exit code 1.
```
<summary>Linting complaints</summary>

</details>

**Review by olleolleolle — APPROVED** [review]
👍 Thank you, @dpep!

**Comment by tilo** [comment]
good work, @dpep! :

**Pull request merged**